File: readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int words_count(string text);
int letters_count(string text);
int sentances_count(string text);

int main(void)
{
    // prompt the user for text
    string text = get_string("Please enter your text: ");

    // counting words
    float words = words_count(text);
    // counting letters
    int letters = letters_count(text);
    // counting sentances
    int sentances = sentances_count(text);

    // l and s
    float L = letters / words * 100;
    float S = sentances / words * 100;

    // index
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int result = round(index);

    // print the result
    if (result < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (result > 1 && result < 16)
    {
        printf("Grade %i\n", result);
    }
    else
    {
        printf("Grade 16+\n");
    }
}

int words_count(string x)
{
    int n = strlen(x);
    char space = ' ';
    int count = 0;
    for (int i = 0; i < n; i++)
    {
        if (x[i] == space)
        {
            count++;
        }
    }
    int words = count + 1;
    return words;
}

int letters_count(string y)
{
    int n = strlen(y);
    char space = ' ';
    char e = 39;
    int count = n;
    for (int i = 0; i < n; i++)
    {
        if (y[i] == space || y[i] == '.' || y[i] == ',' || y[i] == '!' || y[i] == '?' || y[i] == e)
        {
            count--;
        }
    }
    return count;
}

int sentances_count(string z)
{
    int n = strlen(z);
    int count = 0;
    for (int i = 0; i < n; i++)
    {
        if (z[i] == '.' || z[i] == '!' || z[i] == '?')
        {
            count++;
        }
    }
    return count;
}

File: readability.py
def main():
    text = input("Enter your text: ")

    words = words_count(text)
    letters = letters_count(text)
    sentances = sentances_count(text)

    L = letters / words * 100
    S = sentances / words * 100

    index = 0.0588 * L - 0.296 * S - 15.8
    result = round(index)

    if result < 1:
        print("Before Grade 1")
    elif 1 < result < 16:
        print(f"Grade {result}")
    else:
        print("Grade 16+")


def words_count(text):
    n = len(text)
    space = ' '
    count = 0
    for i in range(n):
        if text[i] == space:
            count += 1
    words = count + 1
    return words


def letters_count(text):
    n = len(text)
    space = ' '
    count = n
    for i in range(n):
        if text[i] == space or text[i] == '.' or text[i] == ',' or text[i] == '!' or text[i] == '?' or text[i] == "'":
            count -= 1
    return count


def sentances_count(text):
    n = len(text)
    count = 0
    for i in range(n):
        if text[i] == '.' or text[i] == '!' or text[i] == '?':
            count += 1
    return count


main()
